torchci/pages/api/drci/drci.ts

```diff
@@ -41,25 +52,121 @@ export async function updateDrciComments(octokit: Octokit, prNumber?: string) {
     );
 
     const workflowsByPR = reorganizeWorkflows(recentWorkflows);
+    await addMergeBaseCommits(octokit, workflowsByPR);
     const sevs = getActiveSEVs(await fetchIssuesByLabel("ci: sev"));
+    const masterFlakyJobs = await getFlakyJobs();
+    const baseCommitJobs = await getBaseCommitJobs(workflowsByPR);
 
-    for (const [pr_number, pr_info] of workflowsByPR) {
-        const { pending, failedJobs } = getWorkflowJobsStatuses(pr_info);
+    await Promise.all(
+      Array.from(workflowsByPR.values()).map(async (pr_info) => {
```

Comment:
Can it be refactored to something like `forAllPRWorkflows(values, callback)`?

Reply:
done